record-editor.py

```diff
@@ -533,30 +526,55 @@ def get_photo_id_from_selection( self ):
 
         # get the ID column (which is hidden in the proxy's view) in the
         # original model.
-        photo_id  = self.photosModel.itemFromIndex( index.sibling( index.row(), 0 ) ).text()
+        photo_id = self.photosModel.itemFromIndex( index.sibling( index.row(), self.ID_COLUMN ) ).data()
 
-        return int( photo_id )
+        return photo_id
 
+    @pyqtSlot( int )
     def remove_photo_editor( self, photo_id ):
+        """
+        Slot invoked when a spawned PhotoRecordEditor is closed by the user.
+        The corresponding entry in the tracking dictionary is removed.
+        
+        Takes 1 argument.
+        
+          photo_id - ID corresponding to the photo in the now closed
+                     PhotoRecordEditor.
+                     
+        Returns nothing.
+        """
 
         print( "Removing photo ID={:d} from the edit list.".format( photo_id ) )
         self.photo_record_editors.pop( photo_id, None )
 
+    @pyqtSlot( int )
     def refresh_photo_record( self, photo_id ):
+        """
+        Slot invoked when a photo record has been committed in a 
+        PhotoRecordEditor and whose state needs to be refreshed within
+        our model.
+        
+        Takes 1 argument.
+        
+          photo_id - ID corresponding to the photo just committed.
+                     
+        Returns nothing.
+        """
 
         # update the model's information about this record's state.
         for photo in self.photos:
             if photo["id"] == photo_id:
                 # find this record in the model by it's photo identifier
                 # (there can, and will, only be one) and update it's
                 # processing state.
-                index = self.photosModel.match( self.photosModel.index( 0, 0 ),
-                                                Qt.DisplayRole,
-                                                str( photo_id ),
+                index = self.photosModel.match( self.photosModel.index( 0, self.ID_COLUMN ),
+                                                DATA_ROLE,
+                                                photo_id,
                                                 1,
-                                                Qt.MatchFixedString )[0]
+                                                Qt.MatchExactly )[0]
```

Comment:
I had assumed they were linked though forgot to ask about that.  Good call
on the string vs integer matching - I had forgot about the data type
difference.

On Sun, Aug 21, 2016 at 7:04 PM, swagneto <notifications@github.com> wrote:

> In record-editor.py
> <https://github.com/gthomsen/streetart-analysis/pull/1#discussion_r75606461>
> :
>
> >                                                  1,
> > -                                                Qt.MatchFixedString )[0]
> > +                                                Qt.MatchExactly )[0]
>
> Note that DATA_ROLE is now being provided, which means the QVariant
> storing the ID integer is examined (which got set via
> QStandardItem.setData() in setup_model). I assumed this would be faster and
> more "correct" than string matching against the displayed (despite the ID
> column being hidden) text of the view.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gthomsen/streetart-analysis/pull/1/files/41ae18f71c9958a8c58e39e1eab98c472e99a783#r75606461>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJjQY98B8g2Q29_9CWHJUurp5QE43fkSks5qiOdygaJpZM4JmEfF>
> .
>